iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/schema/ClusterSchemaFetchExecutor.java

```diff
@@ -46,42 +52,54 @@
 import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
-import java.util.function.BiFunction;
 import java.util.function.Consumer;
-import java.util.function.Supplier;
 
 class ClusterSchemaFetchExecutor {
 
+  private final IoTDBConfig config = IoTDBDescriptor.getInstance().getConfig();
   private final Coordinator coordinator;
   private final ITemplateManager templateManager;
-  private final Supplier<Long> queryIdProvider;
-  private final BiFunction<Long, Statement, ExecutionResult> statementExecutor;
+  private final ISchemaFetcher schemaFetcher;
   private final Consumer<ClusterSchemaTree> schemaCacheUpdater;
 
   ClusterSchemaFetchExecutor(
       Coordinator coordinator,
       ITemplateManager templateManager,
-      Supplier<Long> queryIdProvider,
-      BiFunction<Long, Statement, ExecutionResult> statementExecutor,
+      ISchemaFetcher schemaFetcher,
       Consumer<ClusterSchemaTree> schemaCacheUpdater) {
     this.coordinator = coordinator;
     this.templateManager = templateManager;
-    this.queryIdProvider = queryIdProvider;
-    this.statementExecutor = statementExecutor;
+    this.schemaFetcher = schemaFetcher;
     this.schemaCacheUpdater = schemaCacheUpdater;
   }
 
+  private ExecutionResult executionStatement(
+      Long queryId, Statement statement, MPPQueryContext context) {
+    return coordinator.execute(
+        statement,
+        queryId,
+        context == null ? null : context.getSession(),
+        "",
+        ClusterPartitionFetcher.getInstance(),
+        schemaFetcher,
+        context == null || context.getQueryType().equals(QueryType.WRITE)
```

Comment:
write should never  have a timeout.

Reply:
Actually, we have already set the timeout field in `Context` for write to `Long.MAX_VALUE` in Coordinator.

Reply:
> write should never have a timeout.

Yes, the statement must be query here but context may be WRITE type when inserting. So I use default timeout here.